### PR TITLE
fix: to_pyarrow_batches

### LINKED ISSUE #397: `into_backend` does not work for `to_pyarrow_batches`
This will raise because `df.empty` is `True`

```
import letsql as ls
from letsql.expr.relations import (
    into_backend,
)
from letsql.tests.test_into_backend import (
    make_merged,
)


def get_trino_t():
    return (
        ls.trino.connect(database="tpch", schema="sf1")
        .table("orders")
        .cast({"orderdate": "date"})
    )


trino_table = get_trino_t()
db_con = ls.duckdb.connect()
df = (
    get_trino_t().head(10_000)
    .pipe(into_backend, db_con)
    .pipe(make_merged)
    .pipe(ls.to_pyarrow_batches)
    .read_pandas()
)
assert not df.empty
```

### PR DESCRIPTION
**Pull request by mesejo**
closes #397

### CODE REVIEW

File: python/letsql/expr/relations.py [hunk 1]
```diff
@@ -236,3 +163,72 @@ def make_unbound_dt(self):
...
+                        name = f"{v.name}_{next(remote.count)}"
+                        kwargs[k] = MarkedRemoteTable(
+                            name,
+                            schema=v.schema,
+                            source=v.source,
+                            namespace=v.namespace,
+                            remote_expr=v.remote_expr,
+                        )
+
+                        batches = get_batches(remote.remote_expr)
```
Comment (dlovell) [inline python/letsql/expr/relations.py:205]: I think we should call `get_batches` only once

File: python/letsql/expr/relations.py [hunk 2]
```diff
@@ -236,3 +163,72 @@ def make_unbound_dt(self):
...
+        seen_expr[ex] = keep
+        return pa.RecordBatchReader.from_batches(schema, result)
+
+    def replacer(node, _, **kwargs):
+        if isinstance(node, Relation):
+            updated = {}
+            for k, v in list(kwargs.items()):
+                try:
+                    if v in tables:
+                        remote: RemoteTableCounter = tables[v]
```
Comment (dlovell) [inline python/letsql/expr/relations.py:195]: why do we need `RemoteTableCounter`?

File: python/letsql/expr/relations.py (hunk 1 above)
Comment (mesejo) [reply python/letsql/expr/relations.py:205]: We need to register one batch per kwarg, each kwarg representing a reference to the RemoteTable. It is similar to searching for occurrences in the SQL string.

File: python/letsql/expr/relations.py (hunk 2 above)
Comment (mesejo) [reply python/letsql/expr/relations.py:195]: We don't need it, removed it

File: python/letsql/expr/relations.py (hunk 1 above)
Comment (dlovell) [reply python/letsql/expr/relations.py:205]: I'm saying we should invoke the tee'ing only once: we should not tee a tee.

Comment (mesejo) [reply python/letsql/expr/relations.py]: Done

**Review by dlovell — APPROVED** [review]

**Pull request merged**